### DISCUSSION TITLE
Отказ от диапазонов IP в Antizapret

### DISCUSSION THREAD

**Discussion by Dimariqe** [xtrime-ru/antizapret-vpn-docker#77] [General]
1. В связи с уже не малым зоопарком протоколов VPN, а так же в связи с выводом OpenVPN в отдельный контейнер предлагаю удалить возможность переадресации IP адресов из контейнера AntiZapret, оставив только функционал переадресации через DNS.
2. При положительном решении по пункту 1, предлагаю исключить из основного контейнера загрузку исходников из https://bitbucket.org/anticensority/antizapret-pac-generator-light и включить их в состав нашего репозитория, так как патчи будут уж слишком большими.

**Comment by xtrime-ru** [comment]
1. Да, так и планируется. Надо только выбрать контейнер для openvpn (желательно с адмикой) и возможностью управления конфигами.  
2. Исходники оттуда скачиваются, а потом docker заменяет часть файлов нашими в Dockerfile. Пока не вижу смысла отказываться. Но если надо будет - то это не сложно.

**Reply by Dimariqe** [reply]
Я начал перенос контейнера, готового мы не найдем с tunnelblick и dpi patch, придется создать Dockerfile.openvpn, а гуи рассматриваю этот вариант: https://github.com/flant/ovpn-admin
Вопрос: делать ui опцией или уже принудительно?

**Reply by Dimariqe** [reply]
https://github.com/d3vilh/openvpn-ui
Вот этот вариант гуи наверное лучше подойдет, поддерживается, и в отличии от первого имеет авторизацию :)

**Reply by xtrime-ru** [reply]
WG у нас принудительно с gui и никто не жаловался. В крайнем случае пользователь может убрать проброс портов, если ему не нужен GUI.

По вариантам не подскажу - не ресерчил этот вопрос. 

Насчет сборки openvpn из исходников - согласен + возможно потребуется как то генерацию конфигов поправить... Но можно сделать свой форк репо, если хороший GUI и сервер ovpn будут в одном контейнере.

**Reply by xtrime-ru** [reply]
И наверное стоит сразу добавить https://github.com/cbeuw/Cloak, Клиенты amnezia и keenetic поддерживают OpenVPN via Cloak. Работает сильно медленее, но обход блокировок гораздо надежнее.

**Reply by Dimariqe** [reply]
Я планировал добавить https://gitlab.com/openconnect/ocserv на будущее, он уже в релизе 4.2 есть, а cloak только с entware, но тогда проще xray использовать.

**Reply by Dimariqe** [reply]
Но Cloak добавлю, там в принципе ничего сложного, обычный прокси.
